src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -524,51 +516,77 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
-                        executingContext.JobHost = jobHost;
-                        executingContext.Config = config;
+                    // Create a dictionary that will pass the user properties from the Executing filter to the Executed filter
+                    Dictionary<string, object> properties = new Dictionary<string, object>();
 
-                        FunctionExecutedContext executedContext = new FunctionExecutedContext(instance.Id, instance.FunctionDescriptor.FullName, parameterHelper.GetParametersAsDictionary(), properties, logger, functionResult);
-                        executedContext.JobHost = jobHost;
-                        executedContext.Config = config;
+                    // Create the context objects for the filter execution
+                    FunctionExecutingContext executingContext = new FunctionExecutingContext(instance.Id, instance.FunctionDescriptor.FullName, parameterHelper.GetParametersAsDictionary(), properties, logger);
+                    executingContext.JobHost = jobHost;
+                    executingContext.Config = config;
 
-                        ExecutionLogic executionLogic = async (filter, next) =>
-                        {
-                            await filter.OnExecutingAsync(executingContext, functionCancellationTokenSource.Token);
-                            await next;
-                            await filter.OnExecutedAsync(executedContext, functionCancellationTokenSource.Token);
-                        };
+                    FunctionExecutedContext executedContext = new FunctionExecutedContext(instance.Id, instance.FunctionDescriptor.FullName, parameterHelper.GetParametersAsDictionary(), properties, logger);
+                    executedContext.JobHost = jobHost;
+                    executedContext.Config = config;
 
-                        Task function = InvokeAsync(invoker, parameterHelper, timeoutTokenSource, functionCancellationTokenSource,
+                    // Lambda function for function invocation
+                    Func<Task> invokeAsync = async () =>
+                    {
+                        await InvokeAsync(invoker, parameterHelper, timeoutTokenSource, functionCancellationTokenSource,
                             throwOnTimeout, timerInterval, instance);
+                    };
 
-                        ExecutionLogicHandler executionLogicHandler = async (filtersToInvoke, functionToInvoke) =>
+                    // Lambda function that wraps filters around the function
+                    Func<Func<Task>, IFunctionInvocationFilter, Func<Task>> wrapFunctionWithFilter = (innerFunction, filter) =>
+                    {
+                        Func<Task> filteredFunction = async () =>
                         {
-                            List<ExecutionLogic> executionTree = new List<ExecutionLogic>();
-
-                            foreach (IFunctionInvocationFilter filter in filtersToInvoke)
+                            try
```

Comment:
I'm pretty sure we can get the desired behavior without the nested try catch. I also don't see a need for multiple FunctionResults - there should only be one, attached to the `executedContext`, which is overwritten any time a new exception is caught